pkg/download/retry.go

```diff
@@ -24,21 +25,36 @@ const (
 	expRetryN = 7 // how many times we retry the Download
 	expRetryK = time.Second * 3
 	expRetryM = 2
+	writeBufSize = 1024 * 8
 )
 
 // WithRetries retrieves a response body using the specified downloader. Any
 // error returned from d will be retried (and retrieved response bodies will be
 // closed on failures). If the retries do not succeed, the last error is returned.
 //
 // It sleeps in exponentially increasing durations between retries.
-func WithRetries(ctx *log.Context, downloaders []Downloader, sf SleepFunc) (io.ReadCloser, error) {
+func WithRetries(ctx *log.Context, f *File, downloaders []Downloader, sf SleepFunc) (int64, error) {
 	var lastErr error
 	for _, d := range downloaders {
 		for n := 0; n < expRetryN; n++ {
 			ctx := ctx.With("retry", n)
 			status, out, err := Download(ctx, d)
 			if err == nil {
-				return out, nil
+				// server returned status code 200 OK
+				// we have a response body, copy it to the file
+				nBytes, err := io.CopyBuffer(f, out, make([]byte, writeBufSize))
```

Comment:
also can we log the content size of out to GuestAgentExtension and local log? Time the download as well.